test/task.e2e-spec.ts

```diff
@@ -0,0 +1,48 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { INestApplication } from '@nestjs/common';
+import * as request from 'supertest';
+import { TasksModule } from '../src/tasks/tasks.module';
+import { TasksService } from '../src/tasks/tasks.service';
+import { TasksRepository } from '../src/tasks/tasks.repository';
+import { AppModule } from '../src/app.module';
+
+describe('AppController (e2e)', () => {
+  let app: INestApplication;
+  const mockTasksRepository = {};
+
+  beforeEach(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [TasksModule, AppModule],
+      providers: [
+        TasksService,
+        {
+          provide: TasksRepository,
+          useValue: mockTasksRepository,
+        },
+      ],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    await app.init();
+  });
+
+  it('/ (GET)', () => {
+    return request(app.getHttpServer()).get('/tasks').expect(200);
+  });
+
+  it('/ (GETBYID)', () => {
```

Comment:
Instead of getting by id over here, it should be better that we first run a post request and then get. This will free us from the burden of always storing a same id in db.

Reply:
Its a great idea, i did not think about it. Updated now accordingly.